Makefile

```diff
@@ -31,22 +31,26 @@ endif
 
 ifeq ($(ASAN), 1)
 	SANITIZER_FLAG=-DENABLE_ADDRESS_SANITIZER=TRUE -DENABLE_THREAD_SANITIZER=FALSE -DENABLE_UBSAN=FALSE
-endif
-ifeq ($(TSAN), 1)
+else ifeq ($(TSAN), 1)
 	SANITIZER_FLAG=-DENABLE_ADDRESS_SANITIZER=FALSE -DENABLE_THREAD_SANITIZER=TRUE -DENABLE_UBSAN=FALSE
-endif
-ifeq ($(UBSAN), 1)
+else ifeq ($(UBSAN), 1)
 	SANITIZER_FLAG=-DENABLE_ADDRESS_SANITIZER=FALSE -DENABLE_THREAD_SANITIZER=TRUE -DENABLE_UBSAN=TRUE
+else ifeq ($(ASAN), 0)
+	SANITIZER_FLAG=-DENABLE_ADDRESS_SANITIZER=FALSE
+else ifeq ($(TSAN), 0)
+	SANITIZER_FLAG=-DENABLE_THREAD_SANITIZER=FALSE
+else ifeq ($(UBSAN), 0)
+	SANITIZER_FLAG=-DENABLE_UBSAN=FALSE
 endif
```

Comment:
Actually, while we're at it, how do you feel about making UBSAN and TSAN not enable eachother? I think it's really weird behaviour that was to get around problems that this PR fixes.

Reply:
Sure. I hadn't actually noticed that UBSAN is enabling the thread sanitizer too. I thought they were all being enabled exclusively.

Or do you mean to also remove setting them exclusively so that `ASAN=x` is just an alias for `-DENABLE_ADDRESS_SANITIZER=x`?

Reply:
Hm. ASAN and TSAN are incompatible, as are ASAN and UBSAN.

What would be nice would be to be able to specify `UBSAN=1` and have it use UBSAN but not TSAN unless asked, but that's kind of a pain and probably not worth it.

I feel like it makes sense to just be consistent and make `ASAN=x` an alias for `-DENABLE_ADDRESS_SANITIZER=x`, and have users manage what sanitizer they want and don't want in a transparent way.

Reply:
Updated to set and unset them independently.